File: 2024 May/Day 1: 2000. Reverse Prefix of Word [Easy]/sol.py
class Solution(object):
    def reversePrefix(self, word, ch):
        if ch not in word: return word
        j=word.index(ch)
        return word[:j+1][::-1]+word[j+1:]

# Python    
# 8 ms (beats 90.53%)
# 11.68 mb (beats 65.06%)

File: 2024 May/Day 1: 2000. Reverse Prefix of Word [Easy]/sol.cpp
class Solution {
public:
    string reversePrefix(string word, char ch) {
        for (int i=0;i<word.size();i++){
            if (word[i]==ch){
                reverse(word.begin(),word.begin()+i+1);
                break;
            }
        }

        return word;
    }
};

// C++
// 0 ms (beats 100.00%)
// 7.51 mb (beats 28.35%)